src/server/game/Battlegrounds/Zones/BattlegroundWS.cpp

```diff
@@ -142,6 +142,24 @@ void BattlegroundWS::RespawnFlagAfterDrop(TeamId teamId)
     _bgEvents.CancelEvent(BG_WS_EVENT_BOTH_FLAGS_KEPT10);
     _bgEvents.CancelEvent(BG_WS_EVENT_BOTH_FLAGS_KEPT15);
     RemoveAssaultAuras();
+
+    CheckFlagKeeperInArea(teamId == TEAM_ALLIANCE ? TEAM_HORDE : TEAM_ALLIANCE);
+}
+
+void BattlegroundWS::CheckFlagKeeperInArea(TeamId teamId)
+{
+    if (GetStatus() != STATUS_IN_PROGRESS || GetFlagState(teamId) != BG_WS_FLAG_STATE_ON_PLAYER)
```

Comment:
I think you should add brackets here. Shin always says they should be everywhere to avoid weird bugs.